staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -1679,6 +1679,34 @@ type VirtualMachineStatus struct {
 	// RunStrategy tracks the last recorded RunStrategy used by the VM.
 	// This is needed to correctly process the next strategy (for now only the RerunOnFailure)
 	RunStrategy VirtualMachineRunStrategy `json:"runStrategy,omitempty" optional:"true"`
+
+	// VolumeMigrationState contains the information about the volumes set
+	// updates related to the volumeUpdateStrategy
+	VolumeUpdateState *VolumeUpdateState `json:"volumeUpdateState,omitempty" optional:"true"`
+}
+
+type VolumeUpdateState struct {
+	// VolumeMigrationState tracks the information related to the volume migration
+	VolumeMigrationState *VolumeMigrationState `json:"volumeMigrationState,omitempty" optional:"true"`
+}
+
+type VolumeMigrationState struct {
+	// MigratedVolumes lists the source and destination volumes during the volume migration
+	// +listType=atomic
+	// +optional
+	MigratedVolumes []StorageMigratedVolumeInfo `json:"migratedVolumes,omitempty"`
+	// The time the migration action began
+	// +nullable
+	StartTimestamp *metav1.Time `json:"startTimestamp,omitempty"`
+	// The time the migration action ended
+	// +nullable
+	EndTimestamp *metav1.Time `json:"endTimestamp,omitempty"`
+	// Indicates the migration completed
+	Completed bool `json:"completed,omitempty"`
+	// Indicates that the migration succeeded
+	Succeeded bool `json:"succeeded,omitempty"`
```

Comment:
Won't these be available anyway?

Reply:
No if the migration object disappears. Again, these fields are required to persist the information when the VMI isn't available anymore

Reply:
So the question is, why we think the information should stay for volume migration but not for "classic" one? What is specific to the volume migration?

Reply:
The information is necessary because the live migration is triggered by the a live update that changes the vm spec. For regular migrations, you don't have any changes in the VM definition. If something goes wrong with the migration and the VMI disappears you don't have the old volumes information anymore.

Reply:
IOW, the difference with regular migration is that in this case is a copy and if it wasn't successful, this can leave the VM in an inconsistent state.